components/admin_console/team_channel_settings/channel/details/index.ts

```diff
@@ -1,88 +1,90 @@
 import {
+    addChannelMember,
+    deleteChannel,
     getChannel as fetchChannel,
+    getChannelModerations as fetchChannelModerations,
     membersMinusGroupMembers,
     patchChannel,
-    updateChannelPrivacy,
-    getChannelModerations as fetchChannelModerations,
     patchChannelModerations,
-    addChannelMember,
     removeChannelMember,
-    updateChannelMemberSchemeRoles,
-    deleteChannel,
     unarchiveChannel,
+    updateChannelMemberSchemeRoles,
+    updateChannelPrivacy,
 } from 'mattermost-redux/actions/channels';
 import {getTeam as fetchTeam} from 'mattermost-redux/actions/teams';
 
 import {
     getGroupsAssociatedToChannel as fetchAssociatedGroups,
     linkGroupSyncable,
-    unlinkGroupSyncable,
     patchGroupSyncable,
+    unlinkGroupSyncable,
 } from 'mattermost-redux/actions/groups';
 
 import {connect} from 'react-redux';
-
 import {getTeam} from 'mattermost-redux/selectors/entities/teams';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import {ActionFunc} from 'mattermost-redux/types/actions';
 
 import {setNavigationBlocked} from 'actions/admin_actions';
 
-import ChannelDetails from './channel_details';
+import ChannelDetails, {ChannelDetailsActions, ChannelDetailsProps} from './channel_details';
 
-function mapStateToProps(state, props) {
+function mapStateToProps(state: GlobalState, props: ChannelDetailsProps) {
     const config = getConfig(state);
     const guestAccountsEnabled = config.EnableGuestAccounts === 'true';
-    const channelID = props.match.params.channel_id;
+    const channelID = props.channelID;
     const channel = getChannel(state, channelID) || {};
     const team = getTeam(state, channel.team_id) || {};
     const groups = getGroupsAssociatedToChannel(state, channelID);
-    const allGroups = getAllGroups(state, channel.team_id);
     const totalGroups = groups.length;
+    const allGroups = getAllGroups(state);
     const channelPermissions = getChannelModerations(state, channelID);
     const teamScheme = getScheme(state, team.scheme_id);
     return {
+        channelID,
         channel,
         team,
-        allGroups,
-        totalGroups,
         groups,
-        channelID,
+        totalGroups,
+        allGroups,
         channelPermissions,
         teamScheme,
         guestAccountsEnabled,
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: Dispatch) {
     return {
-        actions: bindActionCreators({
-            loadScheme,
-            patchChannelModerations,
-            getChannelModerations: fetchChannelModerations,
-            getChannel: fetchChannel,
-            getTeam: fetchTeam,
+        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc>, ChannelDetailsActions>({
             getGroups: fetchAssociatedGroups,
             linkGroupSyncable,
             unlinkGroupSyncable,
             membersMinusGroupMembers,
-            patchChannel,
             setNavigationBlocked,
```

Comment:
Sorry, I meant casting this function as `any`, not the whole block.
Apologies for the confusion.

Reply:
Oops, sorry, I've been a little bit heavy on the fix :D ! Is it better in 9b3b185de7 ?